API 2.0.0

SirFancyBacon opened the pull request:
Starting the work to migrate to API 2.0.0. definitely don't want to merge this just yet.

But! i am welcoming some guidance as to what is likely wrong with this :)

SirFancyBacon committed:
API 2.0.0

Started work on updating classes/Races to API 2.0.0

NeumimTo commented on thirdparty-config/classes/Race-ExRace.conf:
```diff
@@ -1,10 +1,35 @@
+SkillPointsPerLevel: 1
+#see the wiki for more information about leveling.
+Leveling: {
+    __class__: cz.neumimto.rpg.players.leveling.Exponential
```
cz.neumimto.rpg.players.leveling.Exponential should be "cz.neumimto.rpg.players.leveling.Exponential"

SirFancyBacon replied:
Fixed

NeumimTo commented on thirdparty-config/classes/Race-ExRace.conf:
```diff
@@ -1,10 +1,35 @@
+#"Name" must a string and must be unique betweem races, classes, and skilltrees
+"Name": "ExRace"
+#"ClassType" must be a string and must be defined in the settings.conf file
+"ClassType": "Race"
```
"ClassType": "Race" should be 
ClassType: Race

SirFancyBacon replied:
Fixed

NeumimTo commented on thirdparty-config/classes/Race-ExRace.conf:
```diff
@@ -1,10 +1,35 @@
+# More indepth information can be found on the Ore wiki for NT-RPG
+#
+#"Name" must a string and must be unique betweem races, classes, and skilltrees
+"Name": "ExRace"
```
"Name": "ExRace" 
To
Name: ExRace

SirFancyBacon replied:
Fixed.

SirFancyBacon committed:
More API 2.0.0 migrations

Fixed some issues with naming and added some more notes for the example classes.

Migrated all races and micrafted the example class.

SirFancyBacon committed:
oops

Forgot to remove some quotes..

SirFancyBacon committed:
doc update

SirFancyBacon committed:
API 2.0.0 complete

Should be all set

hellbringer616 commented:
Okay you should be all set.
Review when you have time to make sure it's up to snuff.
